ml-proto/spec/eval.ml

```diff
@@ -88,6 +88,9 @@ let numerics_error at = function
       error at "runtime: invalid conversion to integer"
   | exn -> raise exn
 
+let some_memory c =
+  match c.module_.memory with Some m -> m | _ -> assert false
```

Comment:
I think we still want to distinguish two classes of execution errors (aside
from traps): bugs in the semantics (i.e., the static semantics not being
sound wrt the operational semantics), and bugs in the implementation. The
latter would be assertion failures, but the former should be observed by a
dedicated exception.